src/LinkProviderInterface.php

```diff
@@ -13,9 +13,9 @@ interface LinkProviderInterface
      * The iterable may be an array or any PHP \Traversable object. If no links
      * are available, an empty array or \Traversable MUST be returned.
      *
-     * @return LinkInterface[]|\Traversable
+     * @return iterable[LinkInterface]
```

Comment:
AFAIK the syntax accepted by static analysis tools and IDEs uses `<>`, so this would be `iterable<LinkInterface>` or `iterable<int, LinkInterface>` if we want to be more explicit about what keys are expected.

Reply:
Is it?  PSR-5 isn't clear on that yet, AFAIK.  I'm good with whatever most IDEs support.  Got a link?

Reply:
I couldn't find any link on IDE support but found this mentioned in both [phpstan] (https://medium.com/@ondrejmirtes/generics-in-php-using-phpdocs-14e7301953) and [psalm](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#generic-arrays)

Reply:
Maybe @ondrejmirtes have some references?

Reply:
PhpStorm ("PHPDoc PSR-5 `<generic>` collection type hints"): https://blog.jetbrains.com/phpstorm/2018/09/phpstorm-2018-3-eap-183-2635-12/

Reply:
`iterable[LinkInterface]` will definitely not be understood by any tool. It's either `iterable|LinkInterface[]` legacy syntax from the NetBeans day, or the modern `iterable<LinkInterface>`.

Reply:
`iterable<LinkInterface>` looks closer to the syntax used by other languages with generics support, so I went with that.  Thanks!